internal/stacks/stackruntime/internal/stackeval/component.go

```diff
@@ -221,6 +221,9 @@ func (c *Component) ResultValue(ctx context.Context, phase EvalPhase) cty.Value
 			}
 			elems[string(k)] = inst.ResultValue(ctx, phase)
 		}
+		if len(elems) == 0 {
+			return cty.EmptyObjectVal
+		}
```

Comment:
Whoops, bit surprised we didn't catch this earlier. Would this be a panic at the moment?

Reply:
This sort of pattern is needed for collections to specify the element type for an empty collection, but that isn't needed for an object type. `EmptyObjectVal` is really just a pre-initialized `cty.ObjectVal(nil)` (or a non-nil empty map).

So this is a potential performance optimization but not necessary for correctness.